src/System.IO.Pipelines/src/System/IO/Pipelines/StreamPipeWriter.cs

```diff
@@ -288,7 +308,10 @@ private async ValueTask<FlushResult> FlushAsyncInternal(CancellationToken cancel
                         _head = segment;
                     }
 
-                    await InnerStream.FlushAsync(localToken).ConfigureAwait(false);
+                    if (_bytesBuffered > 0)
```

Comment:
I think this is fine, but it seems fishy to not making calling FlushAsync on the PipeWriter not be 1 to 1 with flushing the stream.

Reply:
I also think this is fine. If you've already successfully flushed the stream after writing every byte that's been buffered so far, I see no point in flushing again.